### ISSUE TITLE
Need a generic application icon

### ISSUE THREAD

**Issue by probonopd** [helloSystem/Artwork#2]
Instead of 

![image](https://user-images.githubusercontent.com/2480569/203854933-8cea413c-8aa3-4301-ae15-548420bc30d0.png)

we need soemething in this style

![image](https://user-images.githubusercontent.com/2480569/203854858-71c227b3-5a48-442d-ba81-2d02669f20d2.png) 
![image](https://user-images.githubusercontent.com/2480569/204150461-703eec7f-a58d-4a51-a9f4-ce5a1ddead08.png)

or  in this style

![image](https://user-images.githubusercontent.com/2480569/203854692-ab10cdb0-931a-4cef-a72d-dfa81e580eaf.png)

so that it can immediately be recognized as being an application.

**Comment by PaulMcClernan** [comment]
![image](https://user-images.githubusercontent.com/15096916/204191031-70c14d93-7c19-4451-a9d5-97a81a352730.png)
Haha! But seriously, I know what you're getting at

**Comment by PaulMcClernan** [comment]
I did a bunch of testing with an helloSys install on an old Core2 Duo rig this weekend, I have some general mac-like usability issues I'd like to make note of, for example command+1 command+2 for icon/list view in Filer doesn't work. Also, please include the macOS 'secret' key-combo command+shift+. (period) to toggle show-invisible files, and command+backspace for move to trash. Command+click window titles path pop-up menu is another 'Mac secret' feature I'm a big fan of.  
I tried some Linux AppImages with many fails (Elf / ABI issue with libQKey...something Qt related to the key swapping? I guess for correcting 'command'-key key, I know the linux OXT app in the AppImage doesn't use Qt). 
Only three out of about 12 AppImages I tried launched OK, I'm guessing its better NOT to include a lot of dependencies inside the AppImage/.AppDir bottle? Or the opposite?
Installed OctoPkg from ports for some GUI pkg management, but that port seems unmaintained.
Installed VLC and seemed to work OK. 
No OpenJDK >11 in FreeBSD 13 ports, I see rayvnOS did a custom build of a newer OpenJDK.
OpenXION (xTalk) ran on OpenJDK11 and did played a music notation sentence with built-in synth/sampler, but text to speech API was not present in that JDK, didn't do much more testing with that, just happy to have some working xTalk scripting present.
Would be helpful to be able to mount some SMB shares.

**Comment by louies0623** [comment]
This one, from the humanity ![humanity-icon-theme](https://user-images.githubusercontent.com/44593430/204239852-c01a37d5-39f4-4235-8aed-c4668bca7cd3.png)

**Comment by PaulMcClernan** [comment]
I just did a quickie rebuild of my OXT icon vector, I was thinking that gears might be better.  
On macOS the executable binaries inside bundles have a different icon (a Terminal), while app bundles have a different default icon (that's changed a lot over the years).  Icons that harken back to macOS Classic may be a little too retro looking after decades+ of Icon designers going in a flattening direction, usually encapsulated in a common shape like rounded rect. May be best, at least for consistency of style, to pick one existing icons theme/style, preferably one that has .svg files available, and then modify or add parts to it as needed. I mean why reinvent the wheel right?

**Comment by PaulMcClernan** [comment]
![generic_app](https://user-images.githubusercontent.com/15096916/204399508-d35e6a27-6d63-401d-9e3a-637be4b76a3e.svg)

**Comment by PaulMcClernan** [comment]
![accessories-text-editor 2-01](https://user-images.githubusercontent.com/15096916/204414489-9c4acb9f-dff7-44cb-9a9e-9f7bfe2ea8f5.svg)

**Comment by PaulMcClernan** [comment]
Should look good in a (future) darkMode

**Comment by PaulMcClernan** [comment]
![accessories-text-editor 2-01](https://user-images.githubusercontent.com/15096916/204416608-58d4197a-6fbd-4a36-b24e-75a405d80c00.svg)

**Comment by PaulMcClernan** [comment]
![accessories-text-editor 2-01](https://user-images.githubusercontent.com/15096916/204417465-330423c8-da38-466d-a2b3-d8ad05e99835.svg)

**Comment by PaulMcClernan** [comment]
![AppImage](https://user-images.githubusercontent.com/15096916/204421034-3bfc5f5e-bbf4-4f9a-9d8f-c38b1dbb9b40.svg)

**Comment by PaulMcClernan** [comment]
![GenericAppIcon](https://user-images.githubusercontent.com/15096916/204421731-ce7fa33a-4e29-4ad6-a2be-690bb269af99.svg)

**Comment by PaulMcClernan** [comment]
![GenericAppMont-01](https://user-images.githubusercontent.com/15096916/204433403-1e4790df-2637-4064-9b25-136ee0aa39a7.svg)

**Comment by PaulMcClernan** [comment]
last one for tonight, loosley based  on Monterey (AppleScript App Bundle) applet icon

**Comment by probonopd** [comment]
Thanks @PaulMcClernan.

Let's go with this one.

https://github.com/helloSystem/Artwork/issues/2#issuecomment-1329964981

Can you try to follow the visual style of https://docs.elementary.io/hig/reference/iconography a bit more closely
and make small icon sizes look really distinctive?

For the generic icon, there should be no text in the document being edited (as it is a generic application, not specifically a text editor).

For a text editor, we might want to use it with the text being edited:

![image](https://user-images.githubusercontent.com/2480569/204612714-23747d87-25f9-46a0-9b70-e86802eb2f23.png)

**Comment by louies0623** [comment]
![image](https://user-images.githubusercontent.com/44593430/204691677-bfb32fe7-adc4-4139-8494-46fff6bd5ec6.png)
You are talking about this icon, right?

**Comment by PaulMcClernan** [comment]
OK I wasn't sure what set you were using as a base/default so will go by Elementary for style going forward.
That generic icon app above @louies0623  looks like it fits the bill?
Seems if you have a hand with a pen/pencil it should be writing something?
Back in the day there were variations of it like MacPaint with hand making a paint swatch.
Can convert to .png at various standard sizes when design is settled.

**Comment by louies0623** [comment]
![generic icon](https://user-images.githubusercontent.com/44593430/204705105-16d35b93-a673-44c3-9bf7-52b6d49131fa.svg)

**Comment by louies0623** [comment]
![generic icon](https://user-images.githubusercontent.com/44593430/204709445-9dabc4b5-baef-41e0-afaa-ebc3576cac63.svg)

**Comment by louies0623** [comment]
![generic icon](https://user-images.githubusercontent.com/44593430/204710345-3806d652-cf96-4d97-b970-0ab48a104bc0.svg)

**Comment by louies0623** [comment]
![generic icon](https://user-images.githubusercontent.com/44593430/204711087-21d535cb-8417-49cf-a757-6f86a0f330b8.svg)

**Comment by PaulMcClernan** [comment]
![GenericAppMont-128-2](https://user-images.githubusercontent.com/15096916/204837819-6b9c9593-0cca-4686-919b-5450882bc71e.png)

**Comment by PaulMcClernan** [comment]
that's 128x128px, I'm sort of stuck on the 'blank slate' idea of 'generic' as application

**Comment by PaulMcClernan** [comment]
![GenericAppMont-32px](https://user-images.githubusercontent.com/15096916/204842197-3aed7a9e-efde-4b23-97f6-7eb7a7445ad5.png)

**Comment by PaulMcClernan** [comment]
Looks OK at 32x32px too, and the light slate coloring stands out nicely on darkMode background.
I hate making bitmap icons smaller than 32px when we all have HD+ screens nowadays, not to mention most modern OS have some sort of scalable vector rendering API available.

**Comment by PaulMcClernan** [comment]
helloSys will have a 'darkMode', right? I mean at least a theme (if not transparency effects and all of that too)

**Comment by PaulMcClernan** [comment]
The piece of paper corner-curling wasn't the case for most app icons in mac sys6-7 classic times, apps had 45 rect (trianglular look)
![macclassicicons](https://user-images.githubusercontent.com/15096916/204848330-971c2b49-70d3-4c55-b2b2-3eeaa8881e75.jpg)

**Comment by louies0623** [comment]
![mac-os-x-snowleopard](https://user-images.githubusercontent.com/44593430/204862363-89b05449-cb49-451d-9dfd-c33395c38d1a.jpg)
I don't like vector rendering icons because it's too much processing power, If these diagrams need to calculate each vector layer, how many have to be calculated, the system is already slow enough.

**Comment by probonopd** [comment]
> helloSys will have a 'darkMode', right?

No. It's violating https://dl.acm.org/doi/book/10.5555/573097

> I don't like vector rendering icons 

helloSystem is using png icons, optimized for the different sizes as per https://docs.elementary.io/hig/reference/iconography#size.

**Comment by louies0623** [comment]
![tumblr_80bbe5f610e25c083045494d4058bed9_342d7561_540](https://user-images.githubusercontent.com/44593430/204869378-c2044269-fc6c-4c96-bfb5-8b877c472233.png)
But if you go back to the black and white icon of system 7, it is simple but hard to get close to

**Comment by probonopd** [comment]
@louies0623 your designs look like document icons to me. In this issue we are looking for a generic application icon. So far I'd say https://github.com/helloSystem/Artwork/issues/2#issuecomment-1329964981 comes closest.

**Comment by PaulMcClernan** [comment]
16px :( yuck
<img width="8" alt="GenericAppMont-16px" src="https://user-images.githubusercontent.com/15096916/204877760-3a1a48e5-f686-4400-a1d1-403f92df931c.png">

**Comment by PaulMcClernan** [comment]
16px

**Comment by PaulMcClernan** [comment]
32px
![GenericAppMont-32px](https://user-images.githubusercontent.com/15096916/204878166-222a9476-d319-4ecb-87d1-559eaf619f72.png)

**Comment by probonopd** [comment]
I think we should rotate the paper by 45 degrees and use no perspective, this way the shape will be distinctive even at very small sizes (blue example below).

![image](https://user-images.githubusercontent.com/2480569/204879594-89dbdced-fb69-4131-98e0-4b3798a4674d.png)

**Comment by PaulMcClernan** [comment]
Yeah but come on! Apple hasn't even followed that HIG even in the 1990s, as soon as there was color screens there were shades of grey or metallic. 

I like darkMode, easier on the eyes IMO and I have this theory that uses less electricity used lighting up laptop LCD pixels.

SVG Icons could be rasterized and then cached, it's only when you get down near 16x16px where you really need human intervention.

As far as processing power for vectors, NeXT did Display Postscript on Moto68K so....? Of course I'm sure that really depends on your vector art, if  your icon has 1000 blends and 5000 points/curves then sure it's going to choke (I've spent 30+ years dealing with that sort of thing on a larger scale)

**Comment by PaulMcClernan** [comment]
> I think we should rotate the paper by 45 degrees and use no perspective, this way the shape will be distinctive even at very small sizes (blue example below).
> 
> ![image](https://user-images.githubusercontent.com/2480569/204879594-89dbdced-fb69-4131-98e0-4b3798a4674d.png)

I was actually working on a "Symbolic" version ( as in this article https://docs.elementary.io/hig/reference/iconography ) a little bit, get it down to a tiny SVG path string (for use in OXT project SVG path widget).
Yes, perspective is problematic, particularly at super-small size, I agree. Also color, note those are 1bit and it actually helps recognition in that use-case.

**Comment by PaulMcClernan** [comment]
![darkSys7](https://user-images.githubusercontent.com/15096916/204883214-c4f84d1c-9519-4c4f-9c99-4844d9fdb7a5.png)

**Comment by louies0623** [comment]
It's just that the icon should retain its original color in dark mode, except black and white.

**Comment by probonopd** [comment]
There is no dark mode in helloSystem.

**Comment by PaulMcClernan** [comment]
Snow Leo probably my favorite Mac look over the years, maybe because that iteration has the most longevity for me so I looked at it the longest.
No DarkMode = Lame, unless you can make it easy enough to add one. Even System 7+ had Appearance Manager, although mostly goofy themes. Might be actually be helpful to break out Sys8 for this sort of work (ResEdit). Just need the option for some darker colored window frames in Window Manager, and reverse to white text.

**Comment by PaulMcClernan** [comment]
![GenericAPP128](https://user-images.githubusercontent.com/15096916/204942807-684a4191-c7d4-41f7-978d-b8b55c131b62.svg)

**Comment by PaulMcClernan** [comment]
SVG (128pt)
![GenericAPP128](https://user-images.githubusercontent.com/15096916/204944578-04543f71-83b7-474f-a96d-68f5d3eeb8c5.svg)

128px PNG
![GenericAPP128](https://user-images.githubusercontent.com/15096916/204944583-482ccbb5-28f2-4f50-bfce-b2efd46ad545.png)

**Comment by PaulMcClernan** [comment]
16x16 pixels
![16-01](https://user-images.githubusercontent.com/15096916/204944620-bd69fd98-d27a-4593-a4ed-a7b3c72c0032.png)

**Comment by PaulMcClernan** [comment]
Symbolic version (Regular & darkMode)
![GenericAppSymbolic](https://user-images.githubusercontent.com/15096916/204944666-97c3d5e6-3aa1-451c-9e29-cf3272b2d408.png)
![GenericAppSymbolicWhite](https://user-images.githubusercontent.com/15096916/204944668-3a53788b-204c-4134-b1c4-9a204e4cc121.png)
SVG 'source':
![GenericAppSymbolic](https://user-images.githubusercontent.com/15096916/204946458-3aeaa854-acc3-492a-9203-a4983ef9dc17.svg)

**Comment by louies0623** [comment]
I agree that there is a dark mode, but I wish it was like elementary os dark mode, not ubuntu dark mode, its color matching is not very comfortable.

**Comment by PaulMcClernan** [comment]
I like the way macOS does 'darkMode' which is more then just changing colors, there's special system NS colors constants you can use instead of RGBA values which are actually dynamic colors and the system does things like changing luminosity of rendered items with the 'vibrant' dark material.

**Comment by PaulMcClernan** [comment]
![16](https://user-images.githubusercontent.com/15096916/204950880-8865eb9e-2526-4341-a6f3-a89168508c3a.svg)
This one follows the Elementary HID specs a little bit closer

**Comment by louies0623** [comment]
![generic icon](https://user-images.githubusercontent.com/44593430/204953448-6408ac10-f42d-438f-a58c-df8792215560.svg)

**Comment by PaulMcClernan** [comment]
NICE! I really like that it has more contrast and generally sharper, maybe better to use the 'hand from that one above it (I cleaned it up some more, has less points).

Still not sure about page curl on a generic app, if we're going by Apple 90s HIG, but even back then there were a bunch of variants, suitcases, extensions, plugins, etc.
I guess we will also need badged variants for icon? Like for the little curved corner Arrow that indicates an alias (symbolic link in our case)?

Anyway I did a full png set of that pervious one, from 128px down to 16px

![GenericAppIconV5-128](https://user-images.githubusercontent.com/15096916/204957583-e180b4f7-4243-401d-93f3-2e625cc6e258.svg)
![GenericAppIconV5-128](https://user-images.githubusercontent.com/15096916/204957584-ab405225-c647-4d18-8027-1f4285336dd5.png)
![GenericAppIconV5-64](https://user-images.githubusercontent.com/15096916/204957586-8aa10e79-3cc2-4184-a61a-e81b5d6f42be.png)
![GenericAppIconV5-32](https://user-images.githubusercontent.com/15096916/204957587-fa6780e0-cfc7-4512-acef-fc876f1df998.png)
![GenericAppIconV5-24](https://user-images.githubusercontent.com/15096916/204957589-2b9d93e1-408f-4c92-8b88-e6148b0e8d5e.png)
![GenericAppIconV5-16px](https://user-images.githubusercontent.com/15096916/204957590-1ebbadac-39c6-4ecf-9045-e51005392d82.png)

**Comment by PaulMcClernan** [comment]
Shopped it in to see how it fits with that Elementary set.
<img width="1096" alt="ElementaryIconsToMatch" src="https://user-images.githubusercontent.com/15096916/204960951-db4b7094-3d7a-4e2a-8ab3-37f5f1ac79eb.png">

**Comment by louies0623** [comment]
![image](https://user-images.githubusercontent.com/44593430/204961259-be2cd29d-e61e-4303-9479-0dafa29a94aa.png)
This is one of the diagrams of MorphOS, maybe you can refer to it, but I just want to ask, is the diagram in their system handed over to Bleander?

**Comment by louies0623** [comment]
My browser displays SVG files incorrectly, but this might work.
![Screenshot_20221201-115214_Chrome](https://user-images.githubusercontent.com/44593430/204962166-d2146600-3662-446e-9276-e9e375e69088.jpg)

**Comment by PaulMcClernan** [comment]
I never used Amiga so I'm not familiar with MorphOS, but checking it out now it may be an option for an up-to-date OS for my old Mirror-Door G4 or even my PowerMac 9360-G4 (9600 with 7300 mobo, don't ask!). 

Calling it quits for tonight.

**Comment by louies0623** [comment]
![204957583-e180b4f7-4243-401d-93f3-2e625cc6e258](https://user-images.githubusercontent.com/44593430/204971510-3aa9a871-4ea6-44b0-b21b-bcfe7036f359.svg)

**Comment by probonopd** [comment]
Thanks for the icons, so far I think
https://github.com/helloSystem/Artwork/issues/2#issuecomment-1333079262
comes closest to what we are after.

This is how it currently looks:

![image](https://user-images.githubusercontent.com/2480569/205136803-a26c2984-593d-47a8-8b7d-9e9d0ea28592.png)

Could we 
* Increase the length of the pencil
* Make the angle of the pencil less steep. Very few people hold a pencil like that ;-)
* Use no earmark for the paper
* Increase the hand a bit and move it down a bit

Let's do some variations around this one.

**Comment by PaulMcClernan** [comment]
OK, will try to work on variations tonight or this weekend. I'm currently building new a OpenXTalk AppImage.

Off topic: 
What desktop/file manager is 'Filer' based on? Files (Nautilus)? Is helloSystems combined File-manger & DE based on Pantheon or what? I want to try editing some of the key-combos.

I just installed Elementary OS on my sons old-old ASUS zen netbook-thing (circa 2016, I think?), and the latest stable release runs very well, considering that it's only DualCore 1.6 ghz Celeron with integrated graphics and only 2GB of ram! Their Pantheon DE may be the most Mac-like experience I've ever had outside of macOS. I changed a few keyboard shortcuts and I almost forgot I was not on macOS! 
As a MacAddict since about 1987, I should really make notes on when somethings I'm expecting is missing from a mac-like experience, could be useful for both elementaryOS and helloSystem. For example command+w closes a window, easy enough, but command+option+w to close ALL of the File manager windows is something I'm always missing on *nix desktops. Option key combo modifiers work with lots for Finder things. For other examples option+clicking-minimize-window-button minimizes ALL Finder windows, command+up-arrow opens a window for it's parent directory but command+option+up-arrow opens a window for it's parent directory AND closes the child-window / current directory, option+disclosure triangle reveals ALL options, there are many examples. These are some of little things that have made macOS 'insanely great' IMO, fast graphical file / window management using key combos. And Consistency, try those in the finder and they work with their System 7 equivalent from three decades ago.

**Comment by probonopd** [comment]
> What desktop/file manager is 'Filer' based on?

It started out as a ork of an older version of PCManFM-Qt but has gone its own ways for a while now.

> As a MacAddict since about 1987, I should really make notes on when somethings I'm expecting is missing from a mac-like experience, could be useful for both elementaryOS and helloSystem. For example command+w closes a window, easy enough, but command+option+w to close ALL of the File manager windows is something I'm always missing on *nix desktops. 

Yes, I'd like to hear about thse things! Would help the project a lot. But each in its own GitHub Ticket please.

> command+up-arrow opens a window for it's parent directory but command+option+up-arrow opens a window for it's parent directory AND closes the window current window.

Command-Shift-Up should work in helloSystem.

**Comment by louies0623** [comment]
![image](https://user-images.githubusercontent.com/44593430/205499535-f3a20f15-f03a-49ec-8fb1-cb394acee22e.png)
https://elementary.io/docs/learning-the-basics#keyboard-shortcuts

**Comment by louies0623** [comment]
The above hand does not have the correct perspective, it should look like this

![image](https://user-images.githubusercontent.com/44593430/205501647-ab3ae217-cd52-4df0-a5d5-e341b43b6d32.png)

**Comment by probonopd** [comment]
Yes, but rotated by 90 degrees to the right.

**Comment by PaulMcClernan** [comment]
![GenericApp4symbolic](https://user-images.githubusercontent.com/15096916/205984755-9b39fd8d-dc31-4bda-9919-0d5e02a454f7.svg)
<img width="16" alt="GenericApp4symbolic-32px" src="https://user-images.githubusercontent.com/15096916/205984758-f2e42084-748e-495d-83f0-905ce669ee2f.png">
<img width="17" alt="GenericApp4-32px" src="https://user-images.githubusercontent.com/15096916/205984759-312a3b2d-cc30-4e2c-ba83-609a08fc7cb0.png">
![GenericApp4](https://user-images.githubusercontent.com/15096916/205984761-398a65e8-1b6a-4556-af3a-07290752e027.svg)

**Comment by PaulMcClernan** [comment]
Sorry to go OT again but I've been wrestling with making this OXT AppImage. it mostly works normally, the only thing broken is OXT runtime ability to dynamically load libraries (it has extensibility, makes bytecode, and like Python it can use libFFI), at least not from within the .AppImages filesystem (which doesn't always appear in the File browsing, although obviously Fuse mounted it) Specifically my OXT FluidSynth wrapper can't load libFluidSynth when asked to. I believe I have the all dependencies in there and executable permissions set. Maybe I need to add some RPATH= for my extensions directories to the AppImage loading process? I could use some advice, and you're the AppImage guy right? Or are you not working on that anymore?

**Comment by probonopd** [comment]
Please make sure that your QXT is loading these things from a path that is calculated by QXT relative to itself. AppImages get mounted at ever-changing locations in the filesystem.

**Comment by PaulMcClernan** [comment]
> Please make sure that your QXT is loading these things from a path that is calculated by QXT relative to itself. AppImages get mounted at ever-changing locations in the filesystem.

Yes I'm noticing that now, that string of random chars in the named mount point, but I believe the toolchain it includes should load it lib relative to the extension module, which is in one of the IDEs sub-directories, but I'll have to double check how the loader finds its modules path wise. Perhaps there's some environmental variable set by AppImage runtime that I could read to get the mount points name once OXT engine is running? I'm going to do some testing / tinkering with it now, I should be able to compile a module from my desktop / mutable local storage outside of the AppImage and install it to the user's App storage directory (OXT has a package manager of sorts).

**Comment by probonopd** [comment]
`$APPDIR`. But please let'S have this discussion elsewhere, as helloSystem != AppImage.

**Comment by PaulMcClernan** [comment]
> `$APPDIR`

Thanks!  I did some more tests and I am getting correct posix paths, and I was able to load Hunspell wrapper and lib.so just fine so the issue might not even be AppImage related.

>But please let'S have this discussion elsewhere, as helloSystem != AppImage.

Agreed, where is good for discussion of AppImage?
